iOS_SDK/OneSignalSDK/Source/OSMessagingController.m

```diff
@@ -122,8 +132,40 @@ - (instancetype)init {
 
 - (void)didUpdateMessagesForSession:(NSArray<OSInAppMessage *> *)newMessages {
     self.messages = newMessages;
-    
+
+    [self resetRedisplayMessagesBySession];
     [self evaluateMessages];
+    [self deleteOldRedisplayedInAppMessages];
+}
+
+- (void)resetRedisplayMessagesBySession {
+    for (NSString *messageId in _redisplayedInAppMessages) {
+        [_redisplayedInAppMessages objectForKey:messageId].isDisplayedInSession = false;
+    }
+}
+
+/*
+ Part of redisplay logic
+ Remove IAMs that the last display time was six month ago
+ */
+- (void)deleteOldRedisplayedInAppMessages {
+    NSMutableSet <NSString *> * messagesIdToRemove = [NSMutableSet new];
```

Comment:
Remove spaces in: `NSMutableSet <NSString *> * messagesIdToRemove = [NSMutableSet new];`

`NSMutableSet <NSString*> *messagesIdToRemove = [NSMutableSet new];`

Reply:
good catch!